### PR TITLE
Add ability to adjust static sampling probabilities per operation

### LINKED ISSUE #814: Can we ignore spans for specific operations?
We do health checks on all our services by having Marathon make a request to GET /external_ping every 5 seconds. These health checks appear as spans in Jaeger and it's quite noisy. Is there a way to avoid have the collector not store these?

screen shot 2018-05-10 at 10 27 25 am

### PR DESCRIPTION
**Pull request by black-adder**
Signed-off-by: Won Jun Jang <wjang@uber.com>

## Which problem is this PR solving?
- ability to adjust sampling probabilities per operation rather than just per service
- resolves #814 

## Short description of the changes
-

### CODE REVIEW

File: scripts/import-order-cleanup.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def main():
     for f in go_files:
         parsed, imports_reordered = parse_go_file(f)
         if output == "stdout" and imports_reordered:
-            print f + " imports out of order"
+            print(f + " imports out of order")
```
Comment (black-adder) [inline scripts/import-order-cleanup.py]: im somehow got python 3 installed on my machine and this is backwards compatible with python 2 anyway

File: plugin/sampling/strategystore/static/strategy.go [hunk 2]
```diff
@@ -21,9 +21,16 @@ type strategy struct {
 	Param float64 `json:"param"`
 }
 
+// operationStrategy defines a operation specific sampling strategy.
```
Comment (yurishkuro) [inline plugin/sampling/strategystore/static/strategy.go]: an operation

**Comment by yurishkuro** [comment]
is there an issue that should be linked to this PR?

**Comment by objectiser** [comment]
Not necessarily in scope for this PR, but I think we will also eventually need to take into account tags - for example, depending upon the instrumentation used, we may get a span reported with `operation=GET` and a tag `http.url=http://xyz/health` - the operation is too generic to isolate the specific endpoint that we want to specify the sampling config for e.g. `/health`.

Reason for raising here is more to do with how the strategy is modeled - it now includes "operation_strategies" - possibly want to make this more generic so could in the future be based on operation and/or a set of tags?

**Comment by yurishkuro** [comment]
@objectiser we've always shied away from trying to infer operation name from the URL, due to REST practice of encoding IDs into the path. The recommendation is to fix the instrumentation to use route as the span name, instead of the http method.

**Pull request merged**